packages/react-server-cli/src/startServer.js

```diff
@@ -160,40 +195,72 @@ const startHotLoadJsServer = (compiler, port, httpsOptions) => {
 		cert: httpsOptions ? httpsOptions.cert : undefined,
 		ca: httpsOptions ? httpsOptions.ca : undefined,
 	});
-	const serverStartedPromise = new Promise((resolve) => {
-		jsServer.listen(port, () => resolve() );
+	const serverStartedPromise = new Promise((resolve, reject) => {
+		jsServer.listen(port, (e) => {
+			if (e) {
+				reject(e);
+				return;
+			}
+			resolve();
+		});
 	});
-	return Promise.all([compiledPromise, serverStartedPromise])
-		.then(() => logger.info(`Started hot reload JavaScript server over ${httpsOptions ? "HTTPS" : "HTTP"} on port ${port}`));
+	return {
+		stop: serverToStopPromise(jsServer),
+		started: Promise.all([compiledPromise, serverStartedPromise])
+			.then(() => logger.info(`Started hot reload JavaScript server over ${httpsOptions ? "HTTPS" : "HTTP"} on port ${port}`)),
+	};
 };
 
+// takes in the err and stats object returned by a webpack compilation and returns
+// an error object if something serious happened, or null if things are ok.
 const handleCompilationErrors = (err, stats) => {
 	if(err) {
 		logger.error("Error during webpack build.");
 		logger.error(err);
-		throw new Error(err);
+		return new Error(err);
```

Comment:
In the past when using a promise library, we just throw, and it rejects automatically. It seems like that's probably not true w/ native promises. In any event, I'm OK with trying out this pattern and changing it if it doesn't work for some reason.